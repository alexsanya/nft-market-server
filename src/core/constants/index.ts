/* eslint-disable @typescript-eslint/no-magic-numbers */

export const SIXTY = 60 as const;
export const ONE_HUNDRED = 100 as const;
export const ONE_THOUSAND = 1000 as const;

export const ZERO = 0 as const;
export const ONE = 1 as const;
export const TEN = 10 as const;

export const ADDRESS_REGEX = new RegExp(/^(0x)?[0-9a-fA-F]{40}$/);
export const BYTES32_REGEX = new RegExp(/^(0x)?[0-9a-fA-F]{64}$/);


export const ETH_MAINNET_ID = "1";
export const POLYGON_ID = "137";
export const SEPOLIA_ID = "11155111";

export const DOMAIN_SEPARATOR_POLYGON = "0x18fe943f718ca9eea10beb465fad697a409bfdab62830c6218e6333db82a854b";
export const DOMAIN_SEPARATOR_SEPOLIA = "0xc011709dbbb9deb3fc6044957fcef85f12c9b999edaa5c6c26250edf4788ecda";

export const DOMAIN_SEPARATORS = {
	[POLYGON_ID]: DOMAIN_SEPARATOR_POLYGON,
	[SEPOLIA_ID]: DOMAIN_SEPARATOR_SEPOLIA
} as Record<string, string>;

export const EIP712_LISTING_TYPES = {
    Listing: [{
        name: "nftContract",
        type: "address"
      },
      {
        name: "tokenId",
        type: "uint256"
      },
      {
        name: "minPriceCents",
        type: "uint256"
      },
      {
        name: "nonce",
        type: "uint256"
      },
    ]
};

export const ERC721_ABI = [
	"function ownerOf(uint256) external view returns (address)"
];

export const ERC20_ABI = [
	"function balanceOf(address) external view returns (uint256)"
];

export enum HttpCode {
	OK = 200,
	CREATED = 201,
	NO_CONTENT = 204,
	BAD_REQUEST = 400,
	UNAUTHORIZED = 401,
	FORBIDDEN = 403,
	NOT_FOUND = 404,
	INTERNAL_SERVER_ERROR = 500
}
