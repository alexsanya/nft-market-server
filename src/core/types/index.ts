export interface ValidationType {
	fields: string[];
	constraint: string;
}

export interface SuccessResponse<T> {
	data?: T;
}

export interface ErrorResponse {
	name: string;
	message: string;
	validationErrors?: ValidationType[];
	stack?: string;
}

export interface Signature {
    v: number;
    r: string;
    s: string;
}