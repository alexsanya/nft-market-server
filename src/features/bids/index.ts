export * from './domain';
export * from './presentation/controller';
export * from './presentation/routes';
