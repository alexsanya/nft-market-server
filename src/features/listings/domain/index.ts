export * from './dtos';
export * from './entities';
export * from './usecases';
export * from './repositories/repository';
export * from './datasources';
