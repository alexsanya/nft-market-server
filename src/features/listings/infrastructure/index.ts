export * from './local.datasource.impl';
export * from './repository.impl';
export * from './onchain.datasource.impl';
export * from './chainDataProvider.impl';
export * from './utils';