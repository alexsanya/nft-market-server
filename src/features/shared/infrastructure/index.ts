export * from './utils';
export * from './redisClient.impl';
export * from './onchain.datasource.impl';
