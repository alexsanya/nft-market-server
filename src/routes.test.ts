import request from 'supertest';
import { envs, HttpCode, SuccessResponse } from './core';
import { testServer } from './testServer';
import { PaginationResponseEntity } from './features/shared';
import { ListingEntity } from './features/listings';

describe('test listing routes', () => {
	const url = `${envs.API_PREFIX}/listings`;

	beforeAll(async () => {
		await testServer.start();
	});

	test('should return listings /listings', async () => {
		await request(testServer.app)
			.get(url)
			.expect(HttpCode.OK)
			.expect('Content-Type', /json/)
			.then(({ body }: { body: SuccessResponse<PaginationResponseEntity<ListingEntity[]>> }) => {
				expect(body).toMatchSnapshot();
				expect(body.data?.results.length).toBe(1);
			});
	});
});
