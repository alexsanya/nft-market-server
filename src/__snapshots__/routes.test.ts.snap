// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test routes listing routes should create new listing /listings 1`] = `
{
  "data": {
    "chainId": "11155111",
    "minPriceCents": "150000",
    "nftContract": "0xf44b599a0aB6b8cb14E992994BEC0dc59dF883B2",
    "nonce": "0",
    "owner": "0x3897326cEda92B3da2c27a224D6fDCFefCaCf57A",
    "signature": {
      "r": "0x5ef4620f4b296763ff15209456d75e868f149a8d1c6821f1ff11fab70bca0ee0",
      "s": "0x337ddcb26ea919a2bf5ad6e1d49bd6951a27d1d2e940a5543a70eabc5dbe237e",
      "v": 28,
    },
    "tokenId": "1",
  },
}
`;

exports[`test routes listing routes should return listings /listings 1`] = `
{
  "data": {
    "currentPage": 1,
    "nextPage": null,
    "prevPage": null,
    "results": [
      {
        "chainId": "11155111",
        "minPriceCents": "150000",
        "nftContract": "0xf44b599a0aB6b8cb14E992994BEC0dc59dF883B2",
        "nonce": "0",
        "owner": "0x3897326cEda92B3da2c27a224D6fDCFefCaCf57A",
        "signature": {
          "r": "0x5ef4620f4b296763ff15209456d75e868f149a8d1c6821f1ff11fab70bca0ee0",
          "s": "0x337ddcb26ea919a2bf5ad6e1d49bd6951a27d1d2e940a5543a70eabc5dbe237e",
          "v": 28,
        },
        "tokenId": "1",
      },
    ],
    "total": 1,
    "totalPages": 1,
  },
}
`;

exports[`test routes test bid routes should create new bid /bids 1`] = `
{
  "data": {
    "bidder": "0xE98D94496aB9084f597a69978b593EBf83147335",
    "listing": {
      "chainId": "11155111",
      "minPriceCents": "150000",
      "nftContract": "0xf44b599a0aB6b8cb14E992994BEC0dc59dF883B2",
      "nonce": "0",
      "owner": "0x3897326cEda92B3da2c27a224D6fDCFefCaCf57A",
      "signature": {
        "r": "0x5ef4620f4b296763ff15209456d75e868f149a8d1c6821f1ff11fab70bca0ee0",
        "s": "0x337ddcb26ea919a2bf5ad6e1d49bd6951a27d1d2e940a5543a70eabc5dbe237e",
        "v": 28,
      },
      "tokenId": "1",
    },
    "signature": {
      "r": "0x1469ac6f9636c24d2d8c3fb2cbef73708876e15f23f23b1d33863939c905a21c",
      "s": "0x7d9a7ea039465c928311bcb737b23153232028038beadba2a667aa720f17602b",
      "v": 28,
    },
    "tokenAddress": "0xc29f6F8D639eF187DcFEfeFBaD989cF2C941a23A",
    "validUntil": "1735504160",
    "value": "250",
  },
}
`;

exports[`test routes test bid routes should return bids /bids 1`] = `
{
  "data": {
    "currentPage": 1,
    "nextPage": null,
    "prevPage": null,
    "results": [
      {
        "bidder": "0xE98D94496aB9084f597a69978b593EBf83147335",
        "listing": {
          "chainId": "11155111",
          "minPriceCents": "150000",
          "nftContract": "0xf44b599a0aB6b8cb14E992994BEC0dc59dF883B2",
          "nonce": "0",
          "owner": "0x3897326cEda92B3da2c27a224D6fDCFefCaCf57A",
          "signature": {
            "r": "0x5ef4620f4b296763ff15209456d75e868f149a8d1c6821f1ff11fab70bca0ee0",
            "s": "0x337ddcb26ea919a2bf5ad6e1d49bd6951a27d1d2e940a5543a70eabc5dbe237e",
            "v": 28,
          },
          "tokenId": "1",
        },
        "signature": {
          "r": "0x1469ac6f9636c24d2d8c3fb2cbef73708876e15f23f23b1d33863939c905a21c",
          "s": "0x7d9a7ea039465c928311bcb737b23153232028038beadba2a667aa720f17602b",
          "v": 28,
        },
        "tokenAddress": "0xc29f6F8D639eF187DcFEfeFBaD989cF2C941a23A",
        "validUntil": "1735504160",
        "value": "250",
      },
    ],
    "total": 1,
    "totalPages": 1,
  },
}
`;
